### PR TITLE
feat: export meta data

### PR DESCRIPTION
**Pull request by n1ru4l**
## Description

As part of https://github.com/kamilkisiela/graphql-hive/issues/1536 we want to store the metadata from graphql-inspector within our database. For this, we need to export the metadata for constructing the messages, as we want to keep them in normalized way and only build the complete detailed reports (with text) at runtime in the business logic.

Context PR with usage of these new APIs: https://github.com/kamilkisiela/graphql-hive/issues/1536

### CODE REVIEW

**Comment by dotansimha** [comment]
@TuvalSimha @kamilkisiela  please review 

@n1ru4l you can rebase to get the release/deps issues fixed

File: .changeset/sour-students-scream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@graphql-inspector/core': minor
+---
+
+include meta data in the changes that are machine processible
```
Comment (dotansimha) [inline .changeset/sour-students-scream.md:5]: Maybe add more details here?

File: packages/core/src/diff/changes/change.ts [hunk 2]
```diff
@@ -73,11 +73,578 @@ export enum CriticalityLevel {
 export interface Criticality {
   level: CriticalityLevel;
   reason?: string;
+  /** Is `true` if the criticality is safe, because of the usage determined via the `checkUsage` config option. */
+  isSafeBasedOnUsage?: boolean;
```
Comment (dotansimha) [inline packages/core/src/diff/changes/change.ts:77]: is it a new thing? should we document it?

File: packages/github/__tests__/diff.test.ts [hunk 3]
```diff
@@ -162,15 +162,15 @@ test('should work with comments and descriptions', async () => {
 test('use interceptor to modify changes', async () => {
   const scope = nock('https://api.com')
     .post('/intercept')
-    .reply(async (_, body: DiffInterceptorPayload) => {
+    .reply((async (_: any, body: DiffInterceptorPayload) => {
```
Comment (dotansimha) [inline packages/github/__tests__/diff.test.ts:165]: any? 😭 😭

File: packages/github/__tests__/diff.test.ts [hunk 4]
```diff
@@ -162,15 +162,15 @@ test('should work with comments and descriptions', async () => {
...
       const response: DiffInterceptorResponse = {
         changes: body.changes.map(c => {
           c.criticality.level = 'NON_BREAKING' as any;
           return { ...c };
         }),
       };
       return [200, response];
-    });
+    }) as any);
```
Comment (dotansimha) [inline packages/github/__tests__/diff.test.ts:173]: any? 😭 😭

**Review by dotansimha — APPROVED** [review]

File: packages/github/__tests__/diff.test.ts (hunk 3 above)
Comment (n1ru4l) [reply packages/github/__tests__/diff.test.ts:165]: 😃

File: packages/core/src/diff/changes/change.ts (hunk 2 above)
Comment (n1ru4l) [reply packages/core/src/diff/changes/change.ts:77]: maybe later

**Pull request merged**